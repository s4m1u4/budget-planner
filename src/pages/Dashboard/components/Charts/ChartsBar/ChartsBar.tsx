import React, { FC } from "react";
import currency from "currency.js";
import {
  BarChart,
  Bar,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ReferenceLine,
  ResponsiveContainer,
} from "recharts";

interface IPayload {
  name: string;
  value: number;
}

interface ICustomTooltip {
  active: boolean;
  payload: IPayload[];
}

interface IChartsBarData {
  name: string | undefined;
  income: number | null;
  expense: number | null;
}

interface ChartsBarProps {
  data: IChartsBarData[];
}

export const ChartsBar: FC<ChartsBarProps> = ({ data }) => {
  const customTooltip = ({ active, payload }: ICustomTooltip | any) => {
    if (active) {
      return (
        <div
          style={{
            color:
              sessionStorage.getItem("theme") === "light" ? "#000" : "#fff",
            padding: "5px",
            backgroundColor:
              sessionStorage.getItem("theme") === "light" ? "#fff" : "#303030",
            border: "1px solid #cccc",
            borderRadius: "6px",
          }}
        >
          <h4>{payload[0].payload.name}</h4>
          {payload.map(
            (data: { name: string; value: number }, index: number) => {
              return (
                <p key={index}>{`${data.name}: ${currency(
                  data.value
                ).format()}`}</p>
              );
            }
          )}
        </div>
      );
    }
    return null;
  };

  return (
    <ResponsiveContainer width="99%" height={330}>
      <BarChart
        width={500}
        height={300}
        data={data}
        margin={{
          top: 5,
          right: 30,
          left: 20,
          bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis dataKey="name" />
        <YAxis />
        <Tooltip content={customTooltip} />
        <Legend iconType="circle" />
        <ReferenceLine y={0} stroke="#000" />
        <Bar dataKey="income" fill="green" />
        <Bar dataKey="expense" fill="red" />
      </BarChart>
    </ResponsiveContainer>
  );
};
