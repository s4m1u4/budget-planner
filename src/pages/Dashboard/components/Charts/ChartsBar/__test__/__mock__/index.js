export { TITLE, DATA } from "./mock";
