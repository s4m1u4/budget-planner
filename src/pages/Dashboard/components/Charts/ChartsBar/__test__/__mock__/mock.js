export const TITLE = "Income";
export const DATA = [
  { name: "Work", income: 17369, expense: -13223 },
  { name: "Travel", income: 74123, expense: -84534 },
  { name: "Education", income: 49763, expense: -2533 },
];
