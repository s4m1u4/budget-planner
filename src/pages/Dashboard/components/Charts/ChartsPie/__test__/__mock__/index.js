export { TITLE, DATA_INCOME } from "./mock";
