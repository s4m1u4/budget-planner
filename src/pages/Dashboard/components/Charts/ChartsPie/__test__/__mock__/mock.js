export const TITLE = "Income";
export const DATA_INCOME = [
  { name: "Work", value: 17369 },
  { name: "Travel", value: 106787 },
  { name: "Education", value: 10456 },
];
