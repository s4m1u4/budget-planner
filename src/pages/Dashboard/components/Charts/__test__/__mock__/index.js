export { CATEGORIES, RECORDS } from "./mock";
