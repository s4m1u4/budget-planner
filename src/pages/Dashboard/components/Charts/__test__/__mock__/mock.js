export const RECORDS = [
  {
    id: "9cc6c0d7-24e3-46aa-aa7c-73e86c1dd230",
    amount: "2354.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "bc732adb-7b27-4d76-944c-4e7013eb97fa",
    amount: "836.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "9410db09-8e1c-4c76-a6b3-e766d9b53ae5",
    amount: "2454.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
];

export const CATEGORIES = [
  {
    id: "084a3558-6b14-4e38-9d24-47e912021384",
    title: "Work",
    description: "Work",
    isDefault: false,
  },
  {
    id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
    title: "Travel",
    description: "awdawda",
    isDefault: false,
  },
  {
    id: "ea298d16-8bdb-4c5b-8198-6b2e5ce2f9a2",
    title: "Education",
    description: "Education it's so important for you!",
    isDefault: false,
  },
];
