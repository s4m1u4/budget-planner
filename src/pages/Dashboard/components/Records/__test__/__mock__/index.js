export { CATEGORIES, LAST_RECORDS } from "./mock";
