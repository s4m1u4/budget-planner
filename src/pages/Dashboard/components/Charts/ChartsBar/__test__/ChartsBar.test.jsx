import { ChartsBar } from "../ChartsBar";
import { render } from "@testing-library/react";
import { DATA } from "./__mock__";

const renderComponent = () => {
  return render(<ChartsBar data={DATA} />);
};

describe("ChartsBar component", () => {
  it("snapshot matching", () => {
    const { baseElement } = renderComponent();
    expect(baseElement).toMatchSnapshot();
  });
});
