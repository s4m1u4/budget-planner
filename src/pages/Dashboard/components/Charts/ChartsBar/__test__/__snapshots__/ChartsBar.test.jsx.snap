// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartsBar component snapshot matching 1`] = `
<body>
  <div>
    <div
      class="recharts-responsive-container"
      style="width: 99%; height: 330px;"
    />
  </div>
</body>
`;
